src/app/searchResults/searchResults.component.js

```diff
@@ -83,6 +83,17 @@ class SearchResultsController {
   productViewDetailsAnalyticsEvent (product) {
     this.analyticsFactory.productViewDetailsEvent(product)
   }
+
+  /**
+   * Go from a full url such as https://give-stage2.cru.org/designations/0/1/2/3/4/0123456.html
+   * to a vanity url of the pattern https://give-stage2.cru.org/0123456
+   * @param originalPath the full url
+   * @returns {*} the vanity url
+   */
+  buildVanity (originalPath) {
+    const pattern = /\/designations\/([0-9])\/([0-9])\/([0-9])\/([0-9])\/([0-9])\/([0-9]{7})\.html/g
```

Comment:
RegEx is ugly no matter what you do, but if you wanted to shorten it, I think you could replace `0-9` with `\d` and `([0-9])\/([0-9])\/([0-9])\/([0-9])\/([0-9])\/` with `(\d\/){5}` and update the match index to `$2`.